src/app/_services/api/problem/submission.service.ts

```diff
@@ -20,7 +21,9 @@ export class SubmissionService {
     }
 
     getPreviousSubmissions(id: number): Observable<QuestionSubmission[]> {
-        const params = new HttpParams().set('question', String(id));
```

Comment:
Services have general use. Maybe another component uses this and doesn't want this ordering. Please add filter options here and pass the submission filter for every use case of this function.
Also, the generalization of the services might be merged before you work on this. Then you have to also modify this accordingly.